gem/view_gem_dag.py

```diff
@@ -0,0 +1,86 @@
+from enum import Enum
+from graphviz import Digraph
+from re import search
+
+class GraphType(Enum):
+    """Type of graph to create
+    PARENT_CHILD: Graph is composed of parent-child relationships only and each node is labelled with all information about the node (its type and all its slots including 'children').
+    DAG: DAG visualisation. Each node label is shortened for readability.
+    GRAPH: The bits of information about each node (i.e. its slots) are unique nodes in the graph. Note that the visualisation from the this case shows the expression graph as a graph rather than as the dag it truely is. This can increase the perceived size of the expression.
+    """
+    PARENT_CHILD = 1
+    DAG = 2
+    GRAPH = 3
+
+
+def view_gem_dag(expressions, filename='gem_dag', directory=None, graph_type=GraphType.GRAPH, format='png'):
```

Comment:
I see that you inherit this API of specifying filename and directory separately, nevertheless, it feels unusual.  Why does Graphviz do it this way?